### PR TITLE
Add type .ico

### PR DESCRIPTION
**Pull request by iwatakeshi**
This adds support for `.ico` files.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -37,6 +37,8 @@ pub enum Type {
     Rgb,
     /// FLIF (Free Lossless Image Format) files
     Flif,
+    /// ICO files
+    ICO
```
Comment (svartalf) [inline src/lib.rs]: Enum variants [should be](https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md#general-naming-conventions) in UpperCamelCase, like an `Ico`

File: src/patterns.rs [hunk 2]
```diff
@@ -16,6 +16,7 @@ const BMP: &'static [u8] = b"BM";
 const BGP: &'static [u8] = b"BPG\xfb";
 const RGB: &'static [u8] = b"\x01\xda";
 const FLIF: &'static [u8] = b"FLIF";
+const ICO: &'static [u8] = b"\x00\x00\x01\x00";
```
Comment (svartalf) [inline src/patterns.rs:19]: Are `.CUR` images still exists, btw? Have not seen them for a decades.

**Review by svartalf — CHANGES_REQUESTED** [review]
Everything else is great, I'll be ready to merge & publish this PR once the naming issue will be addressed.

File: src/patterns.rs (hunk 2 above)
Comment (iwatakeshi) [reply src/patterns.rs:19]: I haven't seen them either. Also, it [isn't listed](https://github.com/image-rs/image#21-supported-image-formats) as one of [image](https://github.com/image-rs/image)'s supported file types.

**Review by svartalf — APPROVED** [review]

**Comment by svartalf** [comment]
Aaaand [published](https://crates.io/crates/imghdr). Thank you for a PR, @iwatakeshi!

**Comment by iwatakeshi** [comment]
@svartalf My pleasure!

**Pull request merged**